config-service-impl/src/main/java/org/hypertrace/config/service/store/InternalContextSpecificConfig.java

```diff
@@ -1,11 +1,29 @@
 package org.hypertrace.config.service.store;
 
+import com.google.protobuf.Value;
 import java.util.Optional;
-import lombok.Value;
 import org.hypertrace.config.service.v1.ContextSpecificConfig;
 
-@Value
 public class InternalContextSpecificConfig {
-  ContextSpecificConfig contextSpecificConfig;
-  Optional<com.google.protobuf.Value> prevConfig;
+  private final ContextSpecificConfig contextSpecificConfig;
+  private final Optional<com.google.protobuf.Value> prevConfigOptional;
+
+  public InternalContextSpecificConfig(ContextSpecificConfig contextSpecificConfig) {
+    this.contextSpecificConfig = contextSpecificConfig;
+    this.prevConfigOptional = Optional.empty();
+  }
+
+  public InternalContextSpecificConfig(
+      ContextSpecificConfig contextSpecificConfig, com.google.protobuf.Value prevConfig) {
+    this.contextSpecificConfig = contextSpecificConfig;
+    this.prevConfigOptional = Optional.of(prevConfig);
+  }
+
+  public ContextSpecificConfig getContextSpecificConfig() {
```

Comment:
You could still leave this as `@Value` with your own constructors, up to you (besides the getters, it will give you equals/hashcode/tostring - important if you're using these anywhere in any of those ways).